torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py

```diff
@@ -226,6 +227,12 @@ def checkpoint_wrapper(
             Wrapped module
     """
 
+    if checkpoint_impl == CheckpointImpl.REENTRANT:
+        warnings.warn(
+            f"Please specify {CheckpointImpl.NO_REENTRANT} as "
+            f"{CheckpointImpl.REENTRANT} will soon be removed as "
+            "the default and eventually deprecated."
```

Comment:
Please add a stack_level arg as we want to turn the inter for that on at some point.